src/functions/assert/Equivalence/Should-BeEquivalent.ps1

```diff
@@ -0,0 +1,774 @@
+        return
+    }
+
+    #compare DataTable
+    if (Is-DataTable -Value $Expected) {
+        # todo add verbose output to data table
+        v "`$Expected is a datatable, we will be comparing `$Actual to datatables."
+        Compare-DataTableEquivalent -Expected $Expected -Actual $Actual -Property $Path -Options $Options
+        return
+    }
+
+    #compare collection
+    if (Is-Collection -Value $Expected) {
+        v "`$Expected is a collection, we will be comparing `$Actual to collections."
+        Compare-CollectionEquivalent -Expected $Expected -Actual $Actual -Property $Path -Options $Options
+        return
+    }
+
+    #compare DataRow
+    if (Is-DataRow -Value $Expected) {
+        # todo add verbose output to data row
+        v "`$Expected is a datarow, we will be comparing `$Actual to datarows."
+        Compare-DataRowEquivalent -Expected $Expected -Actual $Actual -Property $Path -Options $Options
+        return
+    }
+
+    v "`$Expected is an object of type $$Expected.GetType(), we will be comparing `$Actual to objects."
+    Compare-ObjectEquivalent -Expected $Expected -Actual $Actual -Property $Path -Options $Options
+}
+
+function Assert-Equivalent {
+    <#
+    .SYNOPSIS
+    Compares two objects for equivalency, by recursively comparing their properties for equivalency.
+
+    .PARAMETER Actual
+    The actual object to compare.
+
+    .PARAMETER Expected
+    The expected object to compare.
+
+    .PARAMETER Because
+    The reason why the input should be the expected value.
+
+    .PARAMETER Options
+    Options for the comparison. Get-EquivalencyOption function is called to get the default options.
+
+    .PARAMETER StrictOrder
+    If set, the order of items in collections will be compared.
+
+    .EXAMPLE
+    ```powershell
+        $expected = [PSCustomObject] @{
+            Name = "Thomas"
+        }
+
+        $actual = [PSCustomObject] @{
+            Name = "Jakub"
+            Age = 30
+        }
+
+        $actual | Should-BeEquivalent $expected
+    ```
+
+    This will throw an error because the actual object has an additional property Age and the Name values are not equivalent.
+
+    .EXAMPLE
+    ```powershell
+        $expected = [PSCustomObject] @{
+            Name = "Thomas"
+        }
+
+        $actual = [PSCustomObject] @{
+            Name = "Thomas"
+        }
+
+        $actual | Should-BeEquivalent $expected
+    ```
+
+    This will pass because the actual object has the same properties as the expected object and the Name values are equivalent.
+
+    .LINK
+    https://pester.dev/docs/commands/Should-BeEquivalent
+
+    .LINK
+    https://pester.dev/docs/assertions
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Position = 1, ValueFromPipeline = $true)]
+        $Actual,
+        [Parameter(Position = 0, Mandatory)]
+        $Expected,
+        [String]$Because,
+        $Options = (Get-EquivalencyOption),
```

Comment:
`Get-EquivalencyOption` is not exposed

Reply:
will make it into parameters, you are right. 

https://github.com/pester/Pester/issues/2477